IBPSA/Resources/C-Sources/WeeklySchedule.c

```diff
@@ -378,10 +312,18 @@ void* weeklyScheduleInit(const int tableOnFile, const char* name, const double t
           double val;
 
           /* a token has been found on this line before, so we're parsing some numerical data*/
-          if (tokensInLine >= scheduleID->n_cols_in) {
-            weeklyScheduleFreeInit(scheduleID);
-            weeklyScheduleFree(scheduleID);
-            ModelicaFormatError("Too many columns on row %i when reading weekly schedule '%s'.", line, name);
+          if (tokensInFirstLine != 0 && tokensInLine + 1 > tokensInFirstLine){
+              weeklyScheduleFreeInit(scheduleID);
+              weeklyScheduleFree(scheduleID);
+              ModelicaFormatError("Too many columns on data line %i when reading weekly schedule '%s'. %i instead of %i.", line, name, tokensInLine + 1, tokensInFirstLine);
+          }else if (tokensInLine >= n_reservedColumns) { /* This code should only be reached upon passing the first line of data */
+            n_reservedColumns += allocSize;
+            scheduleID->rules[rule_i]->data = (double*)realloc(sizeof(double), n_reservedColumns);
```

Comment:
This is wrong, check the function signature of `realloc`.

Reply:
Good catch.. thanks! I did not realise this code was untested.  I tested it using a long table and spotted another implementation error that is now fixed!